src/routers/handlers/reasonForExtensionHandler.ts

```diff
@@ -53,20 +60,83 @@ export class ReasonForExtensionHandler extends GenericHandler<BaseViewData> {
     }
 
     public async executePost (req: Request, res: Response): Promise<ViewModel<ExtensionReasonViewData>> {
-
         const viewData = await this.getViewData(req, res);
         const selectedOption = req.body?.whyDoYouNeedAnExtension;
         const validator = new PscExtensionsFormsValidator();
         const errorKey: string | null = validator.validateExtensionReason(selectedOption);
 
         if (errorKey) {
             viewData.errors = { whyDoYouNeedAnExtension: { summary: errorKey } };
+            return {
+                templatePath: ROUTER_VIEWS_FOLDER_PATH + PATHS.REASON_FOR_EXTENSION,
+                viewData
+            };
         }
-        return {
-            templatePath: ROUTER_VIEWS_FOLDER_PATH + PATHS.REASON_FOR_EXTENSION,
-            viewData
+
+        logger.info(`called`);
+        // create a new transaction
+        const transaction: Transaction = await postTransaction(req);
+        logger.info(`CREATED transaction with transactionId="${transaction.id}"`);
+
+        // create a new submission for the company number provided
+        const resource = await this.createNewSubmission(req, transaction);
+
+        if (this.isErrorResponse(resource)) {
+
+            const nextPageUrl = PREFIXED_URLS.EXTENSION_REFUSED;
+
+            return {
+                nextPageUrl,
+                templatePath: ROUTER_VIEWS_FOLDER_PATH + PATHS.REASON_FOR_EXTENSION,
+                viewData
+            };
+        } else {
+            const pscExtension = resource.resource;
+            logger.info(`CREATED New Resource ${pscExtension?.links.self}`);
+
+            // set up redirect to confirmation screen
```

Comment:
is this the most clear way to set up the next page url? - the reasonForExtension router then switches it to the Confirmation page.
Couldn't the next page be set here to be `FIRST_EXTENSION_CONFIRMATION` ?
and then the router checks for viewData.errors:
```
    if (!Object.keys(params.viewData.errors).length) {
        res.redirect(params.viewData.nextPageUrl);
    } else {
        res.render(params.templatePath, params.viewData);
    }
```
I haven't tried, maybe I'm not seeing it clearly?

Reply:
so can the `reasonForExtensionRouter.post` be simplified now?